src/components/frame/RouteLayouts/UserNavigation/Wallet/Wallet.tsx

```diff
@@ -88,6 +90,40 @@ const Wallet = () => {
           onClick={handleConnectWallet}
         />
       )}
+      <div className={styles.main}>
+        {wallet?.address && (
+          <GasStationProvider>
+            <GasStationPopover
+              transactionAndMessageGroups={
+                transactionAndMessageGroups as unknown as TransactionOrMessageGroups
+              }
+            >
+              {({ isOpen, toggle, ref }) => (
+                <>
+                  <button
+                    type="button"
+                    className={
+                      isOpen ? styles.walletAddressActive : styles.walletAddress
+                    }
+                    ref={ref}
+                    onClick={toggle}
+                    data-test="gasStationPopover"
+                  >
+                    <span>
+                      <MaskedAddress address={wallet.address as string} />
```

Comment:
Yes. `wallet.address` is marked as `Maybe`, while `MaskedAddress` requires it set. 

This is marked as `Maybe`, although by this point in the process it will be set, otherwise things break way before this. I guess the actual wallet needs to be typed differently to fix the core problem here.

The current types come from `web3-onboard` directly, meaning that if we are to fix this, we need to overwrite them / create our own